simpler_setup/insight_trace/runner.py

```
@@ -0,0 +1,53 @@
+from .models import TraceConfig, TraceResult
+
+
+def run_workspace(config: TraceConfig) -> TraceResult:
+    script = config.output_dir / "run_collect.sh"
+    env = os.environ.copy()
+    if config.cann_home is not None:
+        env["CANN_HOME"] = str(config.cann_home)
+    if config.pto_isa_root is not None:
+        env["PTO_ISA_ROOT"] = str(config.pto_isa_root)
+    env["REPO_ROOT"] = str(config.repo_root)
+    result = subprocess.run(["bash", str(script)], cwd=str(config.output_dir), env=env, check=False, text=True)
```

Ruff (0.15.14)

[error] `subprocess` call: check for execution of untrusted input

(S603)